main.go

```diff
@@ -332,9 +347,197 @@ func mainCore() error {
+		loadHeight := wireDBheight
+		if usePG && (auxDBheight < wireDBheight) {
+			loadHeight = auxDBheight
+		}
+
+		// Fetches the first block hash whose blockdata will be loaded in the
+		// explorer for it pages.
+		loadBlockHash, err := dcrdClient.GetBlockHash(int64(loadHeight))
+		if err != nil {
+			return fmt.Errorf("failed to fetch the block at height (%d), dcrdata dbs must be corrupted",
+				loadHeight)
+		}
+
+		// Signal the goroutine to load this block hash's data.
+		latestBlockHash <- loadBlockHash
+	}
+
+	// Starts a goroutine that fetches the raw updates, process them and
+	// set them as the latest sync status progress updates. The goroutine exits
+	// when no sync is running.
+	explore.SyncStatusUpdate(barLoad)
+
 	blockDataSavers = append(blockDataSavers, explore)
 
-	// Sync up with the blockchain
+	// Register for notifications from dcrd
+	cerr := notify.RegisterNodeNtfnHandlers(dcrdClient)
+	if cerr != nil {
+		return fmt.Errorf("RPC client error: %v (%v)", cerr.Error(), cerr.Cause())
+	}
+
+	// Create the insight socket server and add it to block savers if in pg mode.
+	// Since insightSocketServer is added into the url before even the sync starts,
+	// this implementation cannot be moved to initiateHandlersAndCollectBlocks function.
+	var insightSocketServer *insight.SocketServer
+	if usePG {
+		insightSocketServer, err = insight.NewSocketServer(notify.NtfnChans.InsightNewTxChan, activeChain)
+		if err == nil {
+			blockDataSavers = append(blockDataSavers, insightSocketServer)
+		} else {
+			return fmt.Errorf("Could not create Insight socket.io server: %v", err)
+		}
+	}
+
+	// WaitGroup for the monitor goroutines
+	var wg sync.WaitGroup
+
+	// Start web API
+	app := api.NewContext(dcrdClient, activeChain, &baseDB, auxDB, cfg.IndentJSON)
+	// Start notification hander to keep /status up-to-date
+	wg.Add(1)
+	go app.StatusNtfnHandler(&wg, quit)
+	// Initial setting of db_height. Subsequently, Store() will send this.
+	notify.NtfnChans.UpdateStatusDBHeight <- uint32(wireDBheight)
+
+	apiMux := api.NewAPIRouter(app, cfg.UseRealIP)
+
+	webMux := chi.NewRouter()
+	webMux.With(explore.SyncStatusPageActivation).Group(func(r chi.Router) {
+		r.Get("/", explore.Home)
+		r.Get("/nexthome", explore.NextHome)
+	})
+	webMux.Get("/ws", explore.RootWebsocket)
+	webMux.Get("/favicon.ico", func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, "./public/images/favicon.ico")
+	})
+	cacheControlMaxAge := int64(cfg.CacheControlMaxAge)
+	FileServer(webMux, "/js", http.Dir("./public/js"), cacheControlMaxAge)
+	FileServer(webMux, "/css", http.Dir("./public/css"), cacheControlMaxAge)
+	FileServer(webMux, "/fonts", http.Dir("./public/fonts"), cacheControlMaxAge)
+	FileServer(webMux, "/images", http.Dir("./public/images"), cacheControlMaxAge)
+
+	webMux.With(explore.SyncStatusPageActivation).Group(func(r chi.Router) {
+		r.NotFound(explore.NotFound)
+		r.Mount("/api", apiMux.Mux)
```

Comment:
I missed this until testing, but `r.Mount("/api", apiMux.Mux)` shouldn't be used with the `explore.SyncStatusPageActivation` middleware.  There should be another middleware that either redirects to the api status endpoint (returning a JSON structure indicating sync progress) or returns an appropriate code like 202, along with  the sync status as JSON.

Reply:
Do we do the same thing for the insight api?

Reply:
Yeah they both need to return sensible json, not a web page

Reply:
How about this as the return json?
```json
{ 
 "message":  "blockchain sync is in progress",
 "sync_complete": "1 of 2",
 "syncs": [ { 
       "progress_bar_id": "initial-load",
       "bar_note": "(Full Mode) Syncing stake, base and auxiliary DBs...",
       "best_block_height": 289673,
       "progress": 100.0,
       "time_to_complete": "0 sec",
       "subtitle": "",
      }, { 
       "progress_bar_id": "address-table-sync",
       "bar_note": "Syncing addresses table with spending info...",
       "best_block_height": 1967,
       "progress": 10.0,
       "time_to_complete": "1hr 20 sec",
       "subtitle": "",
      } ],
}
```

Reply:
Parsing it is tricky.  A program might want to know what stage, and stage progress.

So instead of `"sync_complete": "1 of 2"`, perhaps `"stage": 2`. <-- numeric

And instead of `"data"`, `"stages"`.

Then it's obvious and easy to get data for the active stage by indexing the stages array.

Reply:
I have updated the return json above, you can also update it too just in case something else is missing.

Reply:
Lgtm